<launch>
    <!-- 지도 데이터 경로를 설정하는 인자 (기본값 지정) -->
    <arg name="dir_map" default="/media/spcomen21/WD5T/Experiment/LiDAR-100/kitti/07/map/" />
    
    <!-- RViz를 실행하여 시각화 설정 파일을 로드함 -->
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find lidar_100)/rviz_cfg/relocate.rviz"/>

    <!-- load_map 노드를 실행하여 지도 데이터를 로드함 -->
    <node name="load_map" pkg="lidar_100" type="load_map" output="screen"> 
        <!-- 지도 데이터 경로를 설정함 -->
        <param name="PARA_DIR_MAP" value="$(arg dir_map)" />
    </node>

    <!-- relocate 노드를 실행하여 재위치 추정을 수행함 -->
    <node name="relocate" pkg="lidar_100" type="relocate" output="screen"> 
        <!-- 지도 데이터 경로를 설정함 -->
        <param name="PARA_DIR_MAP" value="$(arg dir_map)" />
        <!-- 재위치 추정을 수행할 Z축 최소값 설정 -->
        <param name="PARA_MIN_Z_LOCATE" type="double" value="-0.5" />
        <!-- 재위치 추정을 수행할 Z축 최대값 설정 -->
        <param name="PARA_MAX_Z_LOCATE" type="double" value="50.0" />
    </node>
</launch>
